packages/block-library/src/gallery/edit.js

```diff
@@ -384,7 +382,6 @@ class GalleryEdit extends Component {
 						{ images.length > 1 && (
 							<ColumnsControl
 								label={ __( 'Columns' ) }
-								{ ...MOBILE_CONTROL_PROPS }
```

Comment:
The problem with this sort of spread is that it makes it unclear to me now in this moment of refactoring whether we can remove the `MOBILE_CONTROL_PROPS` object or any of its properties. I see that it is spread into other components, but I have no idea what (or if any) of the properties are being used in those components.

I would have preferred that each prop were individually listed, even if that is more verbose, to make it clearer what is needed and where.

Or better yet, and to the same point as https://github.com/WordPress/gutenberg/pull/20231#issuecomment-586293328, we aim to avoid any differences in the component API, so that we don't need to do this at all.

Reply:
Hi @aduth :wave: 

This current implementation with the `MOBILE_CONTROL_PROPS` object (as well as the even _uglier_ `MOBILE_CONTROL_PROPS_SEPARATOR_NONE` object [introduced here](https://github.com/WordPress/gutenberg/pull/19548#issue-361283348)) is admittedly a pretty hacky workaround. This in part arises from differences in how controls are displayed on web and native (on native, `PanelBody` is in the bottomsheet).

There are at least two areas of technical debt that prompt this awkwardness:

1. Native implementations of various control components have different default `separatorType`s.
2. The responsibility of rendering (or not) the separator currently belongs to the `Cell` component, from which all mobile control components are composed.

#### Proposed solutions:

Due to 1., we must "manually" specify the separator types to ensure consistency in a group of controls (not ideal). Instead, the controls should have a common default `separatorType` obviating the need for this hack. Due to 2., the responsibility of handling fence-post issues for the separators is left to the parent components. It might make more sense to shift this responsibility to the native implementation of `PanelBody`, or if not there, a container component to abstract it away and avoid fence-post logic "leaking" into the parent views.

In the current implementation, these spreadables were extracted to the top per this discussion: https://github.com/WordPress/gutenberg/pull/18265#discussion_r351860385, however, you raise an interesting point about the "painfulness" signalling. Having each prop verbosely listed (with appropriate comments included) could serve to amplify the existence of the underlying technical debt (which is currently signaled only by the code comment).

Reply:
Yeah, I don't exactly understand how this needs to behave in mobile with showing in the bottom card, but it does not seem to me that this separator ought to be a concern of the `*Control` components, which are otherwise labeled input controls agnostic to where they occur in the page. If it's a consequence of how these are rendered within `PanelBody`, then I would agree this should be managed by `PanelBody` (either by styling, or perhaps in its render behavior in inspecting/iterating its `props.children`).